doc/develop/west/alias.rst

```diff
@@ -0,0 +1,55 @@
+.. _west-aliases:
+
+West aliases
+############
+
+West allows to add alias commands to the local, global or system configuration files.
+These aliases make it easy to add shortcuts for frequently used, or hard to memorize
+commands for ease of development.
+
+Similar to how ``git`` aliases work, the alias command is replaced with the alias'
+full text and parsed as a new shell argument list. This enables adding argument
+parameters as they were passed to the original command.
+
+To add a new alias simply call the ``west config`` command:
+
+.. code-block:: console
+
+   west config alias.command "other-command --with-arg"
+
+Recursion
+---------
+
+Recursive aliases are allowed as an alias command can contain other aliases, effectively
+building more complex, but easy to remember commands.
+
+Overriding
+----------
+
+.. warning::
+
+   Overriding/shadowing other or built-in commands is an advanced use case, it can lead to
+   strange side-effects and should be done with great care.
+
+It is possible to override an existing command, for example to pass default argumants:
+
+.. code-block:: console
+
+   west config alias.update "update -o=--depth=1 -n"
+
+Examples
+--------
+
+Add ``west run`` and ``west menuconfig`` shortcuts to your global configuration that
+call ``west build`` with the corresponding CMake targets:
+
+.. code-block:: console
+
+   west config --global alias.run "build --pristine=never --target run"
+   west config --global alias.menuconfig "build --pristine=never --target menuconfig"
```

Comment:
As mentioned in https://github.com/zephyrproject-rtos/west/pull/716#pullrequestreview-2183816277, please avoid overriding built-in commands in the examples:
```suggestion
   west config --global alias.brun "build --pristine=never --target run"
   west config --global alias.menucfg "build --pristine=never --target menuconfig"
```

Reply:
This isn't overriding anything, there currently are no commands `west run` or `west menuconfig`. The actual commands are `west build -t run` and `west build -t menuconfig`.